council/llm/ollama_llm_configuration.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Final, List, Literal, Mapping, Optional, Tuple, Type, Union
+
+from council.utils import Parameter, greater_than_validator, read_env_str, zero_to_one_validator
+
+from . import LLMConfigSpec, LLMConfigurationBase
+
+_env_var_prefix: Final[str] = "OLLAMA_"
+
+
+class OllamaLLMConfiguration(LLMConfigurationBase):
+    def __init__(self, model: str, keep_alive: Optional[Union[float, str]] = None, json_mode: bool = False) -> None:
+        """
+        Initialize a new instance
+
+        Args:
+            model (str): model name to use from https://ollama.com/library
+            keep_alive (Optional[float | str]): ollama keep_alive parameter
+            json_mode (bool): whenever to use json mode, default False
+        """
+        super().__init__()
+        self._model = Parameter.string(name="model", required=True, value=model)
+        keep_alive_value = keep_alive if keep_alive is None else str(keep_alive)
+        self._keep_alive = Parameter.string(name="keep_alive", required=False, value=keep_alive_value)
+        self._json_mode = Parameter.bool(name="json_mode", required=False, value=json_mode)
+
+        # https://github.com/ollama/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values
+        self._mirostat = Parameter.int(name="mirostat", required=False)
+        self._mirostat_eta = Parameter.float(name="mirostat_eta", required=False)
+        self._mirostat_tau = Parameter.float(name="mirostat_tau", required=False)
+        self._num_ctx = Parameter.int(name="num_ctx", required=False, validator=greater_than_validator(0))
+        self._repeat_last_n = Parameter.int(name="repeat_last_n", required=False)
+        self._repeat_penalty = Parameter.float(name="repeat_penalty", required=False)
+        self._temperature = Parameter.float(
+            name="temperature", required=False, default=0.0, validator=zero_to_one_validator
+        )
+        self._seed = Parameter.int(name="seed", required=False)
+        self._stop = Parameter.string(name="stop", required=False)
+        self._tfs_z = Parameter.float(name="tfs_z", required=False)
+        self._num_predict = Parameter.int(name="num_predict", required=False)
+        self._top_p = Parameter.float(name="top_p", required=False, validator=zero_to_one_validator)
+        self._top_k = Parameter.int(name="top_k", required=False, validator=greater_than_validator(0))
+        self._min_p = Parameter.float(name="min_p", required=False)
+
+    def model_name(self) -> str:
+        return self._model.unwrap()
+
+    @property
+    def model(self) -> Parameter[str]:
+        """Ollama model."""
+        return self._model
+
+    @property
+    def keep_alive(self) -> Parameter[str]:
```

Comment:
should'nt we prefer a float or int?

Reply:
No, as it could be both number in seconds and duration string, such as "10m" or "24h"